presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -592,6 +593,52 @@ public Node visitResetSession(SqlBaseParser.ResetSessionContext context)
         return new ResetSession(getLocation(context), getQualifiedName(context.qualifiedName()));
     }
 
+    @Override
+    public Node visitGrant(SqlBaseParser.GrantContext context)
+    {
+        String grantee;
+        if (context.PUBLIC() != null) {
+            grantee = "PUBLIC";
+        }
+        else {
+            grantee = context.grantee.getText();
+        }
+
+        List<String> privileges;
+        if (context.ALL() != null && context.PRIVILEGES() != null) {
+            //List all privileges  declared in 'Privilege' enum in SPI
+            privileges = ImmutableList.of("SELECT", "DELETE", "INSERT");
+        }
+        else {
+            privileges = context.privilege().stream()
+                    .map(privilegeContext -> getPrivilege(privilegeContext))
+                    .collect(toList());
+        }
+        return new Grant(
+                getLocation(context),
+                privileges,
+                context.TABLE() != null,
+                getQualifiedName(context.qualifiedName()),
+                grantee,
+                context.OPTION() != null);
+    }
+
+    private String getPrivilege(SqlBaseParser.PrivilegeContext context)
+    {
+        if (context.SELECT() != null) {
+            return "SELECT";
+        }
+        else if (context.DELETE() != null) {
+            return "DELETE";
+        }
+        else if (context.INSERT() != null) {
+            return "INSERT";
+        }
+        else {
+            throw new IllegalArgumentException("Unsupported Privilege: " + context.getText());
```

Comment:
Done in https://github.com/Teradata/presto/commit/f0ccf281bc9f13601cc1826efe262a57016e1675